actions/watch.ts

```diff
@@ -26,15 +29,35 @@ export const checkForAndPlaceOrder: ActionFn = async (
         "getTradeableOrder()",
         [Array.from(order)]
       );
-      console.log(`Placing Order: ${order}`);
-      await placeOrder(
-        { ...order, from: contract_address, signature },
-        chainContext.api_url
-      );
-    } catch (e) {
+
+      if (order.validTo * 1000 < new Date().getTime()) {
+        console.log(`Invalid order: validTo (${order.validTo}) is in the past `)
+      } else {
+        console.log(`Placing Order: ${order}`);
+        await placeOrder(
+          { ...order, from: contract_address, signature },
+          chainContext.api_url
+        );
+      }
+    } catch (e: any) {
       console.log(`Not tradeable (${e})`);
+      if (e.code === "CALL_EXCEPTION") {
+        contractsToDelete.push(contract_address)
+      }
     }
   }
+
+  for (const contract_address of contractsToDelete) {
+    const contractIndex = registry.contracts.findIndex((existing: string) => existing == contract_address);
+    if (contractIndex >= 0) {
+      registry.contracts.splice(contractIndex, 1);
+      console.log(`Removing contract ${contract_address}`);
+    }
+  }
+  if(contractsToDelete.length > 0) {
+    registry.write();
+  }
```

Comment:
shouldn't we also reset the `contractToDelete`?

Reply:
No, `contractToDelete` doesn't persist between new action executions